src/edgepi/digital_output/edgepi_digital_output.py

```diff
@@ -37,15 +51,22 @@ def digital_output_state(self, pin_name: DoutPins = None, state: bool = None):
             raise ValueError(f'Invalid state passed: {state}')
         if pin_name is None or pin_name.value not in [pins.value for pins in DoutPins]:
             raise InvalidPinName(f'Invalid pin name passed: {pin_name}')
-        if state:
+        if state == DoutTriState.HIGH:
             self.gpio.set_pin_state(pin_name.value)
-        else:
+        elif state == DoutTriState.LOW:
             # In order to safely switch internal MUX circuit, Analog enable pin must be set and
             # cleared with a small time delay. This allows overriding AOUT with DOUT
             self.gpio.set_pin_state(self._dout_aout_pair[pin_name].value)
             self.gpio.clear_pin_state(pin_name.value)
             time.sleep(0.05)
             self.gpio.clear_pin_state(self._dout_aout_pair[pin_name].value)
+        else:
+            # very similar to LOW state, set analog_enable, clear dout and set direction to input
+            # set dac to 0V
+            self.dac.write_voltage(self._dout_dac_pair[pin_name],0)
+            self.gpio.set_pin_state(self._dout_aout_pair[pin_name].value)
+            self.gpio.clear_pin_state(pin_name.value)
+            self.gpio.set_pin_direction_in(pin_name.value)
 
     def digital_output_direction(self, pin_name: DoutPins = None, direction: bool = None):
```

Comment:
please make `digital_output_direction` private and rename it to `set_dout_direction` or something better. functions are better to have actions in their names.

Reply:
digitial output direction should be removed since we are incorporating the Hi-Z state into the set_dout_state() method

Reply:
c64b58fe15dd94139db7e678f51a964dd20c790c